src/main/java/org/pac4j/jax/rs/pac4j/JaxRsProfileManager.java

```diff
@@ -0,0 +1,155 @@
+package org.pac4j.jax.rs.pac4j;
+
+import java.security.Principal;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.SecurityContext;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileHelper;
+import org.pac4j.core.profile.ProfileManager;
+
+/**
+ * 
+ * @author Victor Noel - Linagora
+ * @since 1.0.0
+ *
+ */
+public class JaxRsProfileManager extends ProfileManager<CommonProfile> {
+
+    public JaxRsProfileManager(WebContext context) {
+        super(context);
+
+        ContainerRequestContext requestContext = getJaxRsContext().getRequestContext();
+        SecurityContext original = requestContext.getSecurityContext();
+        if (!(original instanceof Pac4JSecurityContext)) {
+            requestContext.setSecurityContext(new Pac4JSecurityContext(original));
```

Comment:
It is for after logout: the JAX-RS runtime returns a default `SecurityContext` for unauthenticated requests, so I thought it was best to reuse it on logout.